packages/terra-disclosure-manager/src/terra-dev-site/doc/disclosure-manager/DisclosureManagerAPI.3.doc.mdx

```diff
@@ -0,0 +1,178 @@
+   * value may not be honored. If the provided preferredType is not supported by any present disclosure managers,
+   * the root disclosure manager will perform the disclosure using its own disclosure type, regardless of the
+   * preferredType value.
+   */
+  preferredType: PropsTypes.string,
+
+  /**
+   * The explicit `height` and `width` values for the disclosure. These values may not be honored due to the
+   * disclosure type or the available viewport size.
+   * `dimensions` should not be provided if a `size` is specified.
+   */
+  size: {
+    /**
+     * Supported `height` values include: `'240'`, `'420'`, `'600'`, `'690'`, `'780'`, `'870'`, `'960'`,
+     * `'1140'`.
+     */
+    height: PropTypes.number,
+
+    /**
+     * Supported `width` values include: `'320'`, `'480'`, `'560'`, `'640'`, `'800'`, `'960'`, `'1120'`,
+     * `'1280'`, `'1440'`, `'1600'`, `'1760'`, `'1920'`.
+     */
+    width: PropsTypes.number,
+  },
+
+  /**
+   * The String size desired for the disclosure. One of `tiny`, `small`, `medium`, `large`, `huge`, or
+   * `fullscreen`. The functional implementation of this size is determined by the component responsible for
+   * providing the render prop to the DisclosureManager.
+   *   > Note: `size` should not be provided if `dimensions` are specified.
+   */
+  size: PropsTypes.func,
+}).then(() => {
+  // do stuff
+});
+```
+</details>
+
+### Example
+```jsx
+<Button
+  text="Open Modal"
+  onClick={() => {
+    disclosureManager.disclose({
+      preferredType: 'modal',
+      size: 'large',
+      content: {
+        key: 'my-disclosed-content',
+        title: 'My Disclosed Content',
+        component: <DisclosedContent />,
+      }
+    }).then({ dismissDisclosure, afterDismiss } => {
+      this.myContentIsDisclosed = true;
+
+      // This function can be cached and triggered as needed.
+      this.myDismissFunction = dismissDisclosure;
+
+      afterDismiss.then(() => {
+        // The afterDismiss will resolve when the content is dismissed, allowing for
+        // cleanup of local state as needed.
+        this.myContentIsDisclosed = false;
+
+        // If the dismissDisclosure function was cached, make sure it gets cleaned up.
+        this.myDismissFunction = false;
+      })
+    }).catch((e) => {
+      // If this function is executed, the disclosure has been denied. You can use this to
+      // clean up any state that may be reliant on the disclosure occurring.
+    })
+  }}
+/>
+```
+
+## Other APIs
+
+In addition to a `disclose` function, a number of other functions are exposed to components rendered in the disclosure mechanism to manage various segments of the disclosure state. The included functions are:
+
+|Function|Description|
+|---|---|
+|`dismiss`|Allows a component to remove itself from the disclosure stack. If the component is the only element in the disclosure stack, the disclosure is closed.|
```

Comment:
Props table loader?

Reply:
These arn't associated with a component so the props table loader doesn't handle it. (there also aren't prop types for these pieces)